pkgs/development/python-modules/gluonts/default.nix

```diff
@@ -0,0 +1,109 @@
+{ lib
+, buildPythonPackage
+, fetchFromGitHub
+, numpy
+, pandas
+, pydantic
+, tqdm
+, toolz
+, typing-extensions
+, pyarrow
+, torch
+, pytorch-lightning
+, protobuf
+, scipy
+, flask
+, gunicorn
+, orjson
+, prophet
+, s3fs
+, fsspec
+, pytestCheckHook
+, flaky
+, pytest-timeout
+, ujson
+, requests
+, holidays
+, matplotlib
+, plotly
+, cpflows
+, numba
+, statsmodels
+, pythonOlder
+}:
+
+buildPythonPackage rec {
+  # [mxnet] extension is not supported, because mxnet is broken on numpy>=1.24: https://github.com/awslabs/gluonts/issues/2509
+  # [R] extension is ommited for now, feel free to add it
+  pname = "gluonts";
+  version = "0.13.4";
+  format = "pyproject";
+
+  disable = pythonOlder "3.7";
+
+  src = fetchFromGitHub {
+    owner = "awslabs";
+    repo = "gluonts";
+    rev = "v${version}";
+    hash = "sha256-jVZwlhs8v0+VOzFdnfKabNidds3VqrWyn5Xz3FXSWMM=";
+  };
+
+  propagatedBuildInputs = [
+    numpy
+    pandas
+    pydantic
+    tqdm
+    toolz
+    typing-extensions
+    pyarrow
+    torch
+    pytorch-lightning
+    protobuf
+    scipy
+    flask
+    gunicorn
+    orjson
+    prophet
+    s3fs
+    fsspec
+    cpflows
+    numba
+    statsmodels
+    plotly
+    matplotlib
+    holidays
+  ];
```

Comment:
Some of the requirement (https://github.com/awslabs/gluonts/blob/dev/setup.py#L122C1-L131C32) could go into `passthru.optional-dependencies`.

Reply:
Fixed